consensus/polybft/state_sync_manager.go

```diff
@@ -159,9 +159,17 @@ func (s *stateSyncManager) saveVote(msg *TransportMessage) error {
 	valSet := s.validatorSet
 	s.lock.RUnlock()
 
-	if valSet == nil || msg.EpochNumber != epoch {
+	if valSet == nil {
```

Comment:
It's a bit more readable in my opinion:

```suggestion
	if valSet == nil || msg.EpochNumber < epoch || msg.EpochNumber > epoch+1{
	     return nil
	}
	
	if msg.EpochNumber == epoch+1 {
	    if err := s.state.EpochStore.insertEpoch(epoch+1, nil); err != nil {
		return fmt.Errorf("error saving msg vote from a future epoch: %d. Error: %w", epoch+1, err)
	    }
	}
```